detector/calorimeter/HCalTileBarrel_o1_v01_geo.cpp

```diff
@@ -223,12 +223,39 @@ static dd4hep::Ref_t createHCal(dd4hep::Detector& lcdd, xml_det_t xmlDet, dd4hep
   }
   
   // Place envelope (or barrel) volume
-  Volume motherVol = lcdd.pickMotherVolume(hCal);
+  Volume motherVol = lcdd.pickMotherVolume(caloDetElem);
   motherVol.setVisAttributes(lcdd.invisible());
-  PlacedVolume placedHCal = motherVol.placeVolume(envelopeVolume);
-  placedHCal.addPhysVolID("system", hCal.id());
-  hCal.setPlacement(placedHCal);
-  return hCal;
+  PlacedVolume envelopePhysVol = motherVol.placeVolume(envelopeVolume);
+  envelopePhysVol.addPhysVolID("system", xmlDet.id());
+  caloDetElem.setPlacement(envelopePhysVol);
+
+  
+  // Create caloData object
+  auto caloData = new dd4hep::rec::LayeredCalorimeterData;
+  caloData->layoutType = dd4hep::rec::LayeredCalorimeterData::BarrelLayout;
+  caloDetElem.addExtension<dd4hep::rec::LayeredCalorimeterData>(caloData);
+
+  caloData->extent[0] = sensitiveBarrelRmin;
+  caloData->extent[1] = sensitiveBarrelRmax; // or r_max ?
```

Comment:
when you have found out the answer to this question, fix code and remove comment